IOPool/Input/src/PoolSource.h

```diff
@@ -80,6 +81,10 @@ namespace edm {
     bool bypassVersionCheck_;
     int const treeMaxVirtualSize_;
     RunNumber_t setRun_;
+    std::vector<RunNumber_t> setRunNumberForEachLumi_;
```

Comment:
On examination, I think that the strategy pattern would greatly improve the code.
Will do.